src/components/Auth/PasswordInput.vue

```diff
@@ -0,0 +1,84 @@
+<script>
+import { defineComponent } from 'vue';
+
+import BaseTextInput from '../global/BaseTextInput.vue';
+import ViewOffIcon from '@/assets/icons/view-off-icon.svg';
+import ViewOnIcon from '@/assets/icons/view-on-icon.svg';
+
+export default defineComponent({
+  name: 'PasswordInput',
+  components: {
+    BaseTextInput,
+    ViewOnIcon,
+    ViewOffIcon
+  },
+  props: {
+    modelValue: {
+      type: String
+    }
+  },
+  data() {
+    return {
+      isPasswordVisible: false,
+      inputType: 'password',
+      inputValue: ''
+    };
+  },
+  watch: {
+    inputValue() {
+      this.$emit('update:modelValue', this.inputValue);
+      this.$emit('focus');
+    }
+  },
+  methods: {
+    toggleIsPasswordVisible() {
+      this.isPasswordVisible = !this.isPasswordVisible;
+      this.inputType === 'text'
+        ? (this.inputType = 'password')
+        : (this.inputType = 'text');
```

Comment:
I'm not sure which one you find more readable, but this could also be flipped to a single assignment 😄

```suggestion
      this.inputType = this.inputType === 'text'
        ? 'password'
        : 'text';
```

Reply:
That's nice, it didn't occur to me to assign the value conditionally at once, thanks!